### PR TITLE
test: pytest fix strict_parametrization_ids providing required id

### PR DESCRIPTION
**Pull request by blackboxsw**
## Proposed Commit Message
```
test: pytest fix strict_parametrization_ids providing required id

Update test_net to avoid duplicated parametrized test values, drop
fixed xfail decorated test cases.
```

## Additional Context

CI failures due to pytest 9.1.0 and newer
https://github.com/canonical/cloud-init/actions/runs/27560683178/job/81471537568?pr=6902

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/unittests/cmd/test_status.py [hunk 1]
```diff
@@ -936,8 +936,22 @@ def common_mocks(self, mocker):
             # different depending on the ActiveState they are mapped to.
             # Because of this I'm only testing SubState combinations seen
             # in real-world testing (or using "any" string if we dont care).
-            ("activating", "enabled", "start", "123", False),
-            ("activating", "enabled", "start", "123", False),
+            pytest.param(
```
Comment (holmanb) [inline tests/unittests/cmd/test_status.py]: Aren't these two actually duplicates? Do we need both?

Comment (blackboxsw) [reply tests/unittests/cmd/test_status.py]: you are correct. Removed

File: tests/unittests/net/test_dhcp.py [hunk 2]
```diff
@@ -842,9 +842,13 @@ def test_ephemeral_dhcp_setup_network_if_url_connectivity(
...
-        NoDHCPLeaseInterfaceError,
-        NoDHCPLeaseInterfaceError,
-        NoDHCPLeaseMissingDhclientError,
+        pytest.param(
+            NoDHCPLeaseInterfaceError, id="no-dhcp-lease-interface-error"
+        ),
+        pytest.param(
+            NoDHCPLeaseMissingDhclientError,
+            id="no-dhcp-lease-missing-dhclient-error",
+        ),
```
Comment (holmanb) [inline tests/unittests/net/test_dhcp.py]: The fix is removing the duplicate, right?

Are pytest params required here?

File: tests/unittests/config/test_cc_set_passwords.py [hunk 3]
```diff
@@ -580,8 +580,8 @@ def _get_str_class_num(self, str):
             (2, ValueError),
             (3, ValueError),
             (4, 4),
-            (5, 4),
-            (5, 4),
+            pytest.param(5, 4, id="length-5-first"),
+            pytest.param(5, 4, id="length-5-second"),
```
Comment (holmanb) [inline tests/unittests/config/test_cc_set_passwords.py]: Why are two of this test required?

File: tests/unittests/config/test_cc_phone_home.py [hunk 4]
```diff
@@ -54,7 +54,7 @@ def test_no_url(self, m_readurl, caplog):
             (0, -1),
             (1, 0),
             (2, 1),
-            ("2", 1),
+            pytest.param("2", 1, id="tries-str-2"),
```
Comment (holmanb) [inline tests/unittests/config/test_cc_phone_home.py]: This one strikes me as odd - I don't see any duplicates.

Why is this required?

**Review by holmanb — CHANGES_REQUESTED** [review]

File: tests/unittests/config/test_cc_phone_home.py (hunk 4 above)
Comment (blackboxsw) [reply tests/unittests/config/test_cc_phone_home.py]: The separate unittest providing a string "2" instead of int 2 is important to exercise the cast within cc_phone_home. But I don't think we should see "strict parametrized duplicate" errors.  [I believe this is an upstream issue which I have now filed](https://github.com/pytest-dev/pytest/issues/14650).

File: tests/unittests/config/test_cc_set_passwords.py (hunk 3 above)
Comment (blackboxsw) [reply tests/unittests/config/test_cc_set_passwords.py]: There is no need, I've also dropped the extra test case (6, 4) because edge case (5, 4) already covers the truncation at edge.

File: tests/unittests/net/test_dhcp.py (hunk 2 above)
Comment (blackboxsw) [reply tests/unittests/net/test_dhcp.py]: Dropped the pytest.param. It's unnecessary delta and doesn't provide any more context than the error class does.

File: tests/unittests/config/test_cc_ansible.py [hunk 5]
```diff
@@ -355,7 +355,7 @@ def test_schema_validation_deprecations(self, config, error_msg):
                     }
                 },
                 "'playbook_name' is a required property",
-                id="require-url-list",
+                id="require-playbook-name-list",
```
Comment (blackboxsw) [inline tests/unittests/config/test_cc_ansible.py:358]: correct naming for the actual test case.

File: tests/unittests/config/test_cc_phone_home.py [hunk 6]
```diff
@@ -54,7 +54,7 @@ def test_no_url(self, m_readurl, caplog):
             (0, -1),
             (1, 0),
             (2, 1),
-            ("2", 1),
+            pytest.param("2", 1, id="tries-str-type-retries"),
```
Comment (blackboxsw) [inline tests/unittests/config/test_cc_phone_home.py]: Required because of upstream [pytest strict param treatment](https://github.com/pytest-dev/pytest/issues/14650)

File: tests/unittests/config/test_cc_set_passwords.py [hunk 7]
```diff
@@ -581,8 +581,6 @@ def _get_str_class_num(self, str):
             (3, ValueError),
             (4, 4),
             (5, 4),
-            (5, 4),
-            (6, 4),
```
Comment (blackboxsw) [inline tests/unittests/config/test_cc_set_passwords.py:585]: dropped as 6,4 is same edge case as 5,4

**Comment by blackboxsw** [comment]
Ready for review @holmanb. Known pycloudlib failure in build-package-and-run-test due to new release of azure-mgmt-resources breaking changes. See https://github.com/canonical/pycloudlib/pull/506 for more info.

File: tests/unittests/config/test_cc_phone_home.py (hunk 6 above)
Comment (holmanb) [reply tests/unittests/config/test_cc_phone_home.py]: Got it - thanks for filing that issue. That is unfortunate indeed.

Looking at the code without the context of this PR and issue, it's not obvious why this is required.

Comment (blackboxsw) [reply tests/unittests/config/test_cc_phone_home.py]: Added comment about the intent and upstream bug.

**Comment by blackboxsw** [comment]
@holmanb all comments are now resolved

**Review by holmanb — APPROVED** [review]
Thanks!
